### ISSUE TITLE
Undefined references on Linux

### ISSUE THREAD

**Issue by zxey** [revery-ui/revery#330]
I get undefined references to couple of function that glfw3 needs.
The concerning functions are located in Xinerama and Xcursor libraries.
This is unfortunately a problem with reason-glfw which does not specify correct linker flags.

The log when building revery:
```
info esy build 0.5.6 (using package.json)
info building reason-fontkit@2.1.0@d41d8cd9
error: build failed with exit code: 1
  build log:
    # esy-build-package: building: reason-fontkit@2.1.0
    # esy-build-package: pwd: /home/richard/.esy/3__________________________________________________________________/b/reason_fontkit-2.1.0-c3a70403
    # esy-build-package: running: 'dune' 'build' '--root' '.'
             gcc src/fontkit.o
    fontkit.cpp: In function 'hb_font_t* get_font_ot(const char*, int)':
    fontkit.cpp:75:12: warning: ignoring return value of 'size_t fread(void*, size_t, size_t, FILE*)', declared with attribute warn_unused_result [-Wunused-result]
           fread(data, length, 1, file);
           ~~~~~^~~~~~~~~~~~~~~~~~~~~~~
        ocamlopt bin_native/test_fontkit.exe (exit 2)
    (cd _build/default && /home/richard/.esy/3__________________________________________________________________/i/ocaml-4.7.1004-a6ae015e/bin/ocamlopt.opt -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -o bin_native/test_fontkit.exe -I /home/richard/.esy/3__________________________________________________________________/i/ocaml-4.7.1004-a6ae015e/lib/ocaml/threads -I /home/richard/.esy/3__________________________________________________________________/i/opam__s__lwt-opam__c__4.1.0-62ec7e5a/lib/lwt -I /home/richard/.esy/3__________________________________________________________________/i/opam__s__lwt-opam__c__4.1.0-62ec7e5a/lib/lwt/unix -I /home/richard/.esy/3__________________________________________________________________/i/opam__s__ocamlfind-opam__c__1.8.0-eecbed0c/lib/bytes -I /home/richard/.esy/3__________________________________________________________________/i/opam__s__result-opam__c__1.3-1fdcd9f3/lib/result -I /home/richard/.esy/3__________________________________________________________________/i/reason_gl_matrix-0.9.9302-3a6eaf9d/lib/reglm -I /home/richard/.esy/3__________________________________________________________________/i/reason_glfw-3.2.1014-825c4662/lib/reglfw -I src /home/richard/.esy/3__________________________________________________________________/i/opam__s__result-opam__c__1.3-1fdcd9f3/lib/result/result.cmxa /home/richard/.esy/3__________________________________________________________________/i/opam__s__lwt-opam__c__4.1.0-62ec7e5a/lib/lwt/lwt.cmxa /home/richard/.esy/3__________________________________________________________________/i/ocaml-4.7.1004-a6ae015e/lib/ocaml/unix.cmxa /home/richard/.esy/3__________________________________________________________________/i/ocaml-4.7.1004-a6ae015e/lib/ocaml/bigarray.cmxa /home/richard/.esy/3__________________________________________________________________/i/ocaml-4.7.1004-a6ae015e/lib/ocaml/threads/threads.cmxa /home/richard/.esy/3__________________________________________________________________/i/opam__s__lwt-opam__c__4.1.0-62ec7e5a/lib/lwt/unix/lwt_unix.cmxa /home/richard/.esy/3__________________________________________________________________/i/reason_gl_matrix-0.9.9302-3a6eaf9d/lib/reglm/reglm.cmxa /home/richard/.esy/3__________________________________________________________________/i/reason_glfw-3.2.1014-825c4662/lib/reglfw/reglfw.cmxa src/fontkit.cmxa bin_native/.test_fontkit.eobjs/test_fontkit.cmx)
    /usr/local/lib/libglfw3.a(x11_init.c.o): In function `initExtensions':
    x11_init.c:(.text+0x1b3d): undefined reference to `XineramaQueryExtension'
    x11_init.c:(.text+0x1b57): undefined reference to `XineramaIsActive'
    /usr/local/lib/libglfw3.a(x11_init.c.o): In function `_glfwCreateCursorX11':
    x11_init.c:(.text+0x244f): undefined reference to `XcursorImageCreate'
    x11_init.c:(.text+0x257c): undefined reference to `XcursorImageLoadCursor'
    x11_init.c:(.text+0x258c): undefined reference to `XcursorImageDestroy'
    /usr/local/lib/libglfw3.a(x11_monitor.c.o): In function `_glfwPlatformGetMonitors':
    x11_monitor.c:(.text+0x758): undefined reference to `XineramaQueryScreens'
    collect2: error: ld returned 1 exit status
    File "caml_startup", line 1:
    Error: Error during linking
    error: command failed: 'dune' 'build' '--root' '.' (exited with 1)
    esy-build-package: exiting with errors above...
    
  building reason-fontkit@2.1.0
esy: exiting due to errors above
```
There are couple of warnings, but the main concerning lines are:
```
/usr/local/lib/libglfw3.a(x11_init.c.o): In function `initExtensions':
    x11_init.c:(.text+0x1b3d): undefined reference to `XineramaQueryExtension'
    x11_init.c:(.text+0x1b57): undefined reference to `XineramaIsActive'
    /usr/local/lib/libglfw3.a(x11_init.c.o): In function `_glfwCreateCursorX11':
    x11_init.c:(.text+0x244f): undefined reference to `XcursorImageCreate'
    x11_init.c:(.text+0x257c): undefined reference to `XcursorImageLoadCursor'
    x11_init.c:(.text+0x258c): undefined reference to `XcursorImageDestroy'
    /usr/local/lib/libglfw3.a(x11_monitor.c.o): In function `_glfwPlatformGetMonitors':
    x11_monitor.c:(.text+0x758): undefined reference to `XineramaQueryScreens'
    collect2: error: ld returned 1 exit status
```

I've published a simple fix to reason-glfw bryphe/reason-glfw#92 when that gets merged, this library and `reason-fontkit` (it uses reason-glfw as a dependency) library need to get updated reason-glfw in order to get this fix. I've tested this locally with both libraries updated and everything works.

**Comment by bryphe** [comment]
Thanks for the thorough investigation and fix, @zxey ! Just picked up the fix and published new versions of the dependencies - picking it up in #332

**Comment by bryphe** [comment]
Should be fixed now by #332 . I'll close it out, but let me know if I missed anything @zxey 👍
